### ISSUE TITLE
[BUG] Parsing exception of date_range and range aggregation response

### ISSUE THREAD

**Issue by szczepanczykd** [opensearch-project/opensearch-java#369]
### What is the bug?
Cannot parse DateRangeAggregate / RangeAggregate buckets because of missing 'key' mapping:
```JsonParsing Property name 'key' is not in the 'type#name' format. Make sure the request has 'typed_keys' set.```

An example JSON sent by API:
```
{
  "aggregations": {
    "expiry_ranges": {
      "date_range": {
        "field": "expDate",
        "ranges": [
          {
            "from": 1676937600000,
            "key": "from-1-to-2-days",
            "to": 1677110399000
          },
          {
            "from": 1677110400000,
            "key": "from-3-to-4-days",
            "to": 1677283199000
          },
          {
            "from": 1677283200000,
            "key": "from-5-to-6-days",
            "to": 1677455999000
          }
        ]
      }
    }
  },
  "size": 0
}
```

OpenSearch response:
```
{
  "took": 53,
  "timed_out": false,
  "_shards": {
    "total": 1,
    "successful": 1,
    "skipped": 0,
    "failed": 0
  },
  "hits": {
    "total": {
      "value": 6,
      "relation": "eq"
    },
    "max_score": null,
    "hits": []
  },
  "aggregations": {
    "date_range#expiry_ranges": {
      "buckets": [
        {
          "key": "from-1-to-2-days",
          "from": 1676937600000,
          "from_as_string": "2023-02-21T00:00:00.000Z",
          "to": 1677110399000,
          "to_as_string": "2023-02-22T23:59:59.000Z",
          "doc_count": 2
        },
        {
          "key": "from-3-to-4-days",
          "from": 1677110400000,
          "from_as_string": "2023-02-23T00:00:00.000Z",
          "to": 1677283199000,
          "to_as_string": "2023-02-24T23:59:59.000Z",
          "doc_count": 2
        },
        {
          "key": "from-5-to-6-days",
          "from": 1677283200000,
          "from_as_string": "2023-02-25T00:00:00.000Z",
          "to": 1677455999000,
          "to_as_string": "2023-02-26T23:59:59.000Z",
          "doc_count": 2
        }
      ]
    }
  }
}
```

Then we have the parsing error:
`
jakarta.json.stream.JsonParsingException: Property name 'key' is not in the 'type#name' format. Make sure the request has 'typed_keys' set.
`

### How can one reproduce the bug?
Please check this repo: https://github.com/szczepanczykd/opensearch_date_range_aggregation_issue
I've created a small demo for that.

### What is the expected behavior?
The response is parsed and the client returns without JsonParsingException.

### What is your host/environment?
Windows 10
`$ docker-compose -v
Docker Compose version v2.12.0
`
Maven OpenSearch version:
`<opensearch.java.version>2.2.0</opensearch.java.version>`

### Do you have any screenshots?
![image](https://user-images.githubusercontent.com/48957041/220125125-f67dec6c-8e6f-41a5-b089-b943fad3f141.png)

### Do you have any additional context?
The "buckets" array has items with a "key" property which is not configured in the RangeBucket class.
I did some fix with additional integration tests.
I will create PR in the next stage -> PR: https://github.com/opensearch-project/opensearch-java/pull/370

**Comment by szczepanczykd** [comment]
Awaiting release :)

**Comment by dblock** [comment]
Closed via https://github.com/opensearch-project/opensearch-java/pull/370

@VachaShah signed up to doing a release I hear ;))

**Comment by VachaShah** [comment]
> Closed via #370
> 
> @VachaShah signed up to doing a release I hear ;))

Yes tracking the release in #380 :)
